advanced_functionality/working_with_tfrecords/dnn-classifier/train.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import tensorflow as tf
+import time
+
+from os.path import isfile, join
+
+
+parser = argparse.ArgumentParser(
+                    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+# Data and model checkpoints directories
+parser.add_argument('--data_dir', type=str, default='/opt/ml/input',
+                    help='directory containing dataset')
+parser.add_argument('--model_dir', type=str, default='save',
+                    help='directory to store checkpointed models')
+args = parser.parse_args()
+
+
+def parser(serialized_example):
+    """
+    Parses a single tf.Example into image and label tensors.
+    Should return a (data, labels) tuple where data is a features dict
+    """
+    features={
+          'sepal_length': tf.FixedLenFeature([], tf.float32),
+          'sepal_width': tf.FixedLenFeature([], tf.float32),
+          'petal_length': tf.FixedLenFeature([], tf.float32),
+          'petal_width': tf.FixedLenFeature([], tf.float32),
+          'label': tf.FixedLenFeature([], tf.int64),
+    }
+    parsed_features = tf.parse_single_example(serialized_example, features)
+    features = dict({
+        'sepal_length': tf.cast(parsed_features['sepal_length'], tf.float32),
+        'sepal_width': tf.cast(parsed_features['sepal_width'], tf.float32),
+        'petal_length': tf.cast(parsed_features['petal_length'], tf.float32),
+        'petal_width': tf.cast(parsed_features['petal_width'], tf.float32),
+    })
+    label = tf.cast(parsed_features['label'], tf.int64)
+    return features, label
+
+
+def input_fn_train():
+    """Training input function that iterates over TFRecord files in data_dir"""
+    dataset_files = [f for f in os.listdir(args.data_dir) if isfile(join(args.data_dir, f))]
+
+    dataset = tf.data.TFRecordDataset(dataset_files)
+    dataset = dataset.map(parser)
+    dataset = dataset.batch(1)
+    dataset = dataset.repeat()
+
+    iterator = dataset.make_one_shot_iterator()
```

Comment:
you can just return the dataset here.

Reply:
Returning the dataset directly results in a key error here. Google docs and existing examples consume from the dataset by creating an iterator: https://github.com/awslabs/amazon-sagemaker-examples/blob/63cc7fac9bc560e1de1e7e6ed10d4b14cd8a1c11/advanced_functionality/tensorflow_bring_your_own/container/cifar10/cifar10.py#L127-L142